rust/headless-client/src/ipc_service.rs

```diff
@@ -568,16 +575,16 @@ impl<'a> Handler<'a> {
         let _connect_span = telemetry_span!("connect_to_firezone").entered();
 
         assert!(self.session.is_none());
-        let device_id = device_id::get_or_create().map_err(|e| Error::DeviceId(e.to_string()))?;
+        let device_id = device_id::get_or_create().context("Failed to get-or-create device ID")?;
 
         let url = LoginUrl::client(
-            Url::parse(api_url).map_err(|e| Error::UrlParse(e.to_string()))?,
+            Url::parse(api_url).context("Failed to parse URL")?,
```

Comment:
Will this still show up in the client logs? Just thinking that we'll probably get users who change this accidentally to something invalid and wonder what's up.

Reply:
Yes, it will show up. We'll see it too in Sentry.